In [1]:
import pandas as pd
import numpy as np

In [2]:
DiabetesDS = pd.read_csv(r'C:\Users\brian\Desktop\Springboard\Capstone Two - Project Ideas\dataset_diabetes\diabetic_data.csv')

In [3]:
DiabetesDS.head()

encounter_id  patient_nbr             race  gender      age weight  \
0       2278392      8222157        Caucasian  Female   [0-10)      ?   
1        149190     55629189        Caucasian  Female  [10-20)      ?   
2         64410     86047875  AfricanAmerican  Female  [20-30)      ?   
3        500364     82442376        Caucasian    Male  [30-40)      ?   
4         16680     42519267        Caucasian    Male  [40-50)      ?   

   admission_type_id  discharge_disposition_id  admission_source_id  \
0                  6                        25                    1   
1                  1                         1                    7   
2                  1                         1                    7   
3                  1                         1                    7   
4                  1                         1                    7   

   time_in_hospital  ... citoglipton insulin  glyburide-metformin  \
0                 1  ...          No      No                   No   
1                 3  ...          No      Up                   No   
2                 2  ...          No      No                   No   
3                 2  ...          No      Up                   No   
4                 1  ...          No  Steady                   No   

   glipizide-metformin  glimepiride-pioglitazone  metformin-rosiglitazone  \
0                   No                        No                       No   
1                   No                        No                       No   
2                   No                        No                       No   
3                   No                        No                       No   
4                   No                        No                       No   

   metformin-pioglitazone  change diabetesMed readmitted  
0                      No      No          No         NO  
1                      No      Ch         Yes        >30  
2                      No      No         Yes         NO  
3                      No      Ch         Yes         NO  
4                      No      Ch         Yes         NO  

[5 rows x 50 columns]

In [4]:
DiabetesDS.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 101766 entries, 0 to 101765
Data columns (total 50 columns):
 #   Column                    Non-Null Count   Dtype 
---  ------                    --------------   ----- 
 0   encounter_id              101766 non-null  int64 
 1   patient_nbr               101766 non-null  int64 
 2   race                      101766 non-null  object
 3   gender                    101766 non-null  object
 4   age                       101766 non-null  object
 5   weight                    101766 non-null  object
 6   admission_type_id         101766 non-null  int64 
 7   discharge_disposition_id  101766 non-null  int64 
 8   admission_source_id       101766 non-null  int64 
 9   time_in_hospital          101766 non-null  int64 
 10  payer_code                101766 non-null  object
 11  medical_specialty         101766 non-null  object
 12  num_lab_procedures        101766 non-null  int64 
 13  num_procedures            101766 non-null  int64 
 14  num_

In [5]:
missing = pd.concat([DiabetesDS.isnull().sum(), 100 * DiabetesDS.isnull().mean()], axis=1)
missing.columns=['count', '%']
missing.sort_values(by='count')

count    %
encounter_id                  0  0.0
chlorpropamide                0  0.0
glimepiride                   0  0.0
acetohexamide                 0  0.0
glipizide                     0  0.0
glyburide                     0  0.0
tolbutamide                   0  0.0
pioglitazone                  0  0.0
rosiglitazone                 0  0.0
acarbose                      0  0.0
miglitol                      0  0.0
troglitazone                  0  0.0
tolazamide                    0  0.0
examide                       0  0.0
citoglipton                   0  0.0
insulin                       0  0.0
glyburide-metformin           0  0.0
glipizide-metformin           0  0.0
glimepiride-pioglitazone      0  0.0
metformin-rosiglitazone       0  0.0
metformin-pioglitazone        0  0.0
change                        0  0.0
nateglinide                   0  0.0
repaglinide                   0  0.0
metformin                     0  0.0
A1Cresult                     0  0.0
patient_nbr                   0  0.0
race                          0  0.0
gender                        0  0.0
age                           0  0.0
weight                        0  0.0
admission_type_id             0  0.0
discharge_disposition_id      0  0.0
admission_source_id           0  0.0
time_in_hospital              0  0.0
payer_code                    0  0.0
diabetesMed                   0  0.0
medical_specialty             0  0.0
num_procedures                0  0.0
num_medications               0  0.0
number_outpatient             0  0.0
number_emergency              0  0.0
number_inpatient              0  0.0
diag_1                        0  0.0
diag_2                        0  0.0
diag_3                        0  0.0
number_diagnoses              0  0.0
max_glu_serum                 0  0.0
num_lab_procedures            0  0.0
readmitted                    0  0.0

In [29]:

missing = pd.concat([DiabetesDS.isin(['?']).sum(), 100 * DiabetesDS.isin(['?']).mean()], axis=1)
missing.columns=['count', '%']
missing.sort_values(by='count')

count          %
encounter_id                  0   0.000000
chlorpropamide                0   0.000000
glimepiride                   0   0.000000
acetohexamide                 0   0.000000
glipizide                     0   0.000000
glyburide                     0   0.000000
tolbutamide                   0   0.000000
pioglitazone                  0   0.000000
rosiglitazone                 0   0.000000
acarbose                      0   0.000000
miglitol                      0   0.000000
troglitazone                  0   0.000000
tolazamide                    0   0.000000
examide                       0   0.000000
citoglipton                   0   0.000000
insulin                       0   0.000000
glyburide-metformin           0   0.000000
glipizide-metformin           0   0.000000
glimepiride-pioglitazone      0   0.000000
metformin-rosiglitazone       0   0.000000
metformin-pioglitazone        0   0.000000
change                        0   0.000000
nateglinide                   0   0.000000
repaglinide                   0   0.000000
metformin                     0   0.000000
A1Cresult                     0   0.000000
patient_nbr                   0   0.000000
gender                        0   0.000000
age                           0   0.000000
admission_type_id             0   0.000000
discharge_disposition_id      0   0.000000
admission_source_id           0   0.000000
time_in_hospital              0   0.000000
diabetesMed                   0   0.000000
num_lab_procedures            0   0.000000
readmitted                    0   0.000000
num_medications               0   0.000000
number_outpatient             0   0.000000
number_emergency              0   0.000000
number_inpatient              0   0.000000
number_diagnoses              0   0.000000
max_glu_serum                 0   0.000000
num_procedures                0   0.000000
diag_1                       21   0.020636
diag_2                      358   0.351787
diag_3                     1423   1.398306
race                       2273   2.233555
payer_code                40256  39.557416
medical_specialty         49949  49.082208
weight                    98569  96.858479

In [7]:
DiabetesDS['patient_nbr'].nunique()

71518

In [16]:
DiabetesDS['admission_type_id'].value_counts()

1    53990
3    18869
2    18480
6     5291
5     4785
8      320
7       21
4       10
Name: admission_type_id, dtype: int64

code 6 is null so 5291 of these records do not have a value.
code 8 is not mapped, so 320 of these records do not have a value.

In [28]:
DDI = DiabetesDS['discharge_disposition_id'].value_counts()
DDI.sort_index()

1     60234
2      2128
3     13954
4       815
5      1184
6     12902
7       623
8       108
9        21
10        6
11     1642
12        3
13      399
14      372
15       63
16       11
17       14
18     3691
19        8
20        2
22     1993
23      412
24       48
25      989
27        5
28      139
Name: discharge_disposition_id, dtype: int64

18 is null = 3691 records